internal/messaging/cheque_handler.go

```diff
@@ -0,0 +1,245 @@
+	previousChequeModel, err := session.GetLatestChequeRecord(ctx, chequeRecordID)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to get previous cheque: %w", err)
+	}
+
+	counter := previousChequeModel.Counter.Add(previousChequeModel.Counter, big.NewInt(1))
+
+	lastCashIn, err := cm.cmAccountInstance.GetLastCashIn(&bind.CallOpts{}, fromBot, toBot)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to get last cash in: %w", err)
+	}
+	if lastCashIn.LastCounter.Cmp(counter) > 0 {
+		return cheques.SignedCheque{}, fmt.Errorf("invalid cheque counter: counter is less than last cash in counter from sc")
+	}
+
+	bigIntAmount := new(big.Int).SetUint64(amount)
+	totalAmount := new(big.Int).Add(previousChequeModel.Amount, bigIntAmount)
+
+	if totalAmount.Cmp(lastCashIn.LastAmount) < 0 {
+		// Return an error indicating that the amount is invalid
+		return cheques.SignedCheque{}, fmt.Errorf("total amount (%s) is smaller than last cash-in amount (%s)", totalAmount.String(), lastCashIn.LastAmount.String())
+	}
+
+	newCheque := &cheques.Cheque{
+		FromCMAccount: fromCMAccount,
+		ToCMAccount:   toCMAccount,
+		ToBot:         toBot,
+		Counter:       counter,
+		Amount:        bigIntAmount,
+		CreatedAt:     big.NewInt(time.Now().Unix()),
+		ExpiresAt:     big.NewInt(0),
+	}
+	signer, err := cheques.NewSigner(cm.privateKey, cm.chainID)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to create signer: %v", err)
+	}
+	signedCheque, err := signer.SignCheque(newCheque)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to sign cheque: %v", err)
+	}
+	previousCheque := &cheques.Cheque{
+		FromCMAccount: previousChequeModel.FromCMAccount,
+		ToCMAccount:   previousChequeModel.ToCMAccount,
+		ToBot:         previousChequeModel.ToBot,
+		Counter:       previousChequeModel.Counter,
+		Amount:        previousChequeModel.Amount,
+		CreatedAt:     previousChequeModel.CreatedAt,
+		ExpiresAt:     previousChequeModel.ExpiresAt,
+	}
+
+	previousChequeSigned, err := signer.SignCheque(previousCheque)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to sign previous cheque: %v", err)
+	}
+
+	verified := cheques.VerifyCheque(previousChequeSigned, signedCheque, big.NewInt(time.Now().Unix()), big.NewInt(0))
+	if verified != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to verify cheque: %v", verified)
+	}
+
+	sc, err := cm.verifyWithSmartContract(ctx, *signedCheque)
+	if err != nil {
+		return cheques.SignedCheque{}, fmt.Errorf("failed to verify cheque with smart contract: %v", err)
+	}
+
+	// insert into db
+	// UpsertChequeRecord(ctx, models.ChequeRecord{
+	// 	ChequeRecordID: chequeRecordID,
+	// 	TxID:           nil,
+	// 	Status:         ChequeTxStatusPending,
+	// 	signedCheque:   signedCheque,
+	// })
+
+	return *signedCheque, nil
+}
+
+func (cm *evmChequeHandler) verifyWithSmartContract(ctx context.Context, signedCheque cheques.SignedCheque) (struct {
```

Comment:
if you'r not using first return value (result struct), than maybe don't return it at all?